packages/cdktf-cli/src/bin/cmds/helper/terraform-cloud-client.ts

```diff
@@ -117,9 +160,14 @@ async function post(url: string, token: string, data: string) {
   });
 }
 
+async function endpointUrl(tfeHostname: string, path: string) {
+  const serviceDiscovery = await discoverService(tfeHostname);
+  return `https://${tfeHostname}${serviceDiscovery["tfe.v2"]}${path}`;
```

Comment:
The serviceDiscovery map values will include `https` and the hostname as part of the value.
https://developer.hashicorp.com/terraform/internals/remote-service-discovery#discovery-process
```suggestion
  return `${serviceDiscovery["tfe.v2"]}${path}`;
```

Reply:
On further inspection of that page it is actually either absolute or relative, and should follow at least 1 redirect:

> This URL may be either absolute or relative, and if relative it is resolved against the final discovery URL (after following redirects).

Reply:
In my testing, they were all relative paths, but you're right, the code should be more defensive.